File: 2444. Count Subarrays With Fixed Bounds/Solution2444.py
class Solution2444:
    def countSubarrays(self, nums: List[int], minK: int, maxK: int) -> int:
        res, ibad, imax, imin = 0, -1, -1, -1
        for i, num in enumerate(nums):
            if num < minK or num > maxK:
                ibad = i
            if num == minK:
                imin = i
            if num == maxK:
                imax = i
            res += max(0, min(imin, imax) - ibad)
        return res

File: 2444. Count Subarrays With Fixed Bounds/Solution2444.cpp
class Solution2444 {
public:
    long long countSubarrays(vector<int>& nums, int minK, int maxK) {
        long res = 0, ibad = -1, imin = -1, imax = -1, n = nums.size();
        for (int i = 0; i < n; i++) {
            if (nums[i] < minK || nums[i] > maxK) ibad = i;
            if (nums[i] == minK) imin = i;            
            if (nums[i] == maxK) imax = i;
            res += max(0L, min(imin, imax) - ibad);
        }
        return res;
    }
};